ui/src/app/pages/analysis/cohort-resource-card.tsx

```diff
@@ -0,0 +1,179 @@
+import {RenameModal} from 'app/components/rename-modal';
+import {Action, ResourceCardTemplate} from 'app/components/resource-card-template';
+import {withConfirmDeleteModal, WithConfirmDeleteModalProps} from 'app/components/with-confirm-delete-modal';
+import {withErrorModal, WithErrorModalProps} from 'app/components/with-error-modal';
+import {withSpinnerOverlay, WithSpinnerOverlayProps} from 'app/components/with-spinner-overlay';
+import {cohortsApi} from 'app/services/swagger-fetch-clients';
+import colors from 'app/styles/colors';
+import {formatRecentResourceDisplayDate} from 'app/utils';
+import {navigateByUrl} from 'app/utils/navigation';
+import {ResourceType} from 'app/utils/resourceActions';
+import {RecentResource} from 'generated/fetch';
+import * as fp from 'lodash';
+import * as React from 'react';
+
+interface Props extends WithConfirmDeleteModalProps, WithErrorModalProps, WithSpinnerOverlayProps {
+  resource: RecentResource;
+  existingNameList: string[];
+  onUpdate: Function;
+}
+
+interface State {
+  showRenameModal: boolean;
+}
+
+export const CohortResourceCard = fp.flow(
+  withErrorModal(),
+  withConfirmDeleteModal(),
+  withSpinnerOverlay(),
+)(class extends React.Component<Props, State> {
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      showRenameModal: false
+    };
+  }
+
+  get resourceType(): ResourceType {
+    return ResourceType.COHORT;
+  }
+
+  get displayName(): string {
+    return this.props.resource.cohort.name;
+  }
+
+  // this is duplicated
+  get writePermission(): boolean {
```

Comment:
I wish we could have these components have reference to a `WorkspaceAccessLevel` so we can just use `WorkspacePermissionUtil` for this. That's a refactor for another day, though.